persistence/src/main/java/de/cyface/persistence/PersistenceLayer.java

```diff
@@ -772,53 +777,85 @@ private Cursor loadEventsCursor(final long measurementIdentifier) {
                 continue;
             }
             Validate.notNull(pauseEventTime);
-
-            // Add all GeoLocations until the pause event as a sub-track
             final long resumeEventTime = eventCursor.getLong(eventCursor.getColumnIndex(EventTable.COLUMN_TIMESTAMP));
-            final Track track = new Track();
-            while (geoLocationCursor.moveToNext()) {
-
-                // Collect locations before pause event
-                GeoLocation location = loadGeoLocation(geoLocationCursor);
-                if (location.getTimestamp() <= pauseEventTime) {
-                    track.add(location);
-                    continue;
-                }
-
-                // Pause reached: Move geoLocationCursor to the first location of the next sub-track
-                // We do this to ignore locations between pause and resume event (STAD-140)
-                while (location.getTimestamp() < resumeEventTime) {
-                    geoLocationCursor.moveToNext();
-                    // Load next location to see if it's the first location of the next sub-track
-                    location = loadGeoLocation(geoLocationCursor);
-                }
-
-                // We reached the first location of the next sub-track, thus, move pointer back one step
-                // As the next outer while iteration will move the pointer one step forward
-                geoLocationCursor.moveToPrevious();
-                break; // Continue with next sub track (i.e. search for next resume event)
-            }
 
+            // Collect all GeoLocations until the pause event
+            final Track track = collectNextSubTrack(geoLocationCursor, pauseEventTime);
             // Add sub-track to track
             if (track.getGeoLocations().size() > 0) {
                 tracks.add(track);
             }
+
+            // Pause reached: Move geoLocationCursor to the first location of the next sub-track
+            // We do this to ignore locations between pause and resume event (STAD-140)
+            moveCursorToFirstAfter(geoLocationCursor, resumeEventTime);
+
+            // Stop GeoLocation collection if we already reached the last GeoLocation
+            if (geoLocationCursor.isAfterLast()) {
```

Comment:
This is weird. This is the last command in the while loop. You could just "or" connect this within the while condition, couldn't you?

Reply:
Done that.